### PR TITLE
Fix: Correct fluentd prase TimeFormat config key

### LINKED ISSUE #778: bug: Parser - TimeFormat is written to time_type
### Describe the issue

I tried adding a parser as filter using timeType and timeFormat to parse the time out of my logs.
Unfortunately the operator writes the timeFormat value as `time_type` to the config.

https://github.com/fluent/fluent-operator/blob/15478961c1fc0ba5e1760f826faf386b094468b0/apis/fluentd/v1alpha1/plugins/common/common_types.go#L201-L203

### To Reproduce

Use a filter like following:
```
---
apiVersion: fluentd.fluent.io/v1alpha1
kind: Filter
metadata:
  name: xyz
  labels:
    filter.fluentd.fluent.io/enabled: "true"
spec:
  filters:
    - parser:
        keyName: log
        parse:
          type: json
          timeType: '*string'
          timeFormat: '%FT%TZ'
```

Ends up like:
```
<parse>
  @type "json"
  time_type %FT%TZ
</parse>
```

### Expected behavior

The fluentd config should set the filed as `time_format` and not as `time_type`.

### Your Environment

```markdown
- Fluent Operator version: 2.3.0
- Container Runtime: containerd
- Operating system: Ubuntu 22.04
```

### How did you install fluent operator?

_No response_

### Additional context

_No response_

### LINKED ISSUE #781: bug: TimeType wants *string instead of string
### Describe the issue

K8S validation prevents users from setting `time_type` to string. It allows `*string` instead.
This results in fluentd crashing with following log line:
```
2023-06-07 11:36:40 +0000 [error]: config error file="/fluentd/etc/fluent.conf" error_class=Fluent::ConfigError error="valid options are string,unixtime,float,mixed but got *string"
```

### To Reproduce

Apply following filter:
```
---
apiVersion: fluentd.fluent.io/v1alpha1
kind: Filter
metadata:
  name: xyz
  labels:
    filter.fluentd.fluent.io/enabled: "true"
spec:
  filters:
    - parser:
        keyName: log
        parse:
          type: json
          timeType: '*string'
          utc: true

```

When setting timeType to the value `string` the validation fails with following line:
```
The Filter "xyz" is invalid: spec.filters[0].parser.parse.timeType: Unsupported value: "string": supported values: "float", "unixtime", "*string", "mixed"
```

### Expected behavior

The k8s validation should allow string instead of *string.

### Your Environment

```markdown
- Fluent Operator version:
- Container Runtime:
- Operating system:
- Kernel version:
```

### How did you install fluent operator?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Jakob3xD**
### What this PR does / why we need it:
It changes the TimeFormat config value from `time_type` to `time_format` as it should be.

### Which issue(s) this PR fixes:
Fixes #778
Fixes #781 

### Does this PR introduced a user-facing change?

```release-note
Fix: Correct fluentd prase TimeFormat config key
Fix: Correct Kubebuilder validation for TimeType from *string to string
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: apis/fluentd/v1alpha1/plugins/common/common_types.go [hunk 1]
```diff
@@ -20,10 +20,10 @@ type CommonFields struct {
 
 // Time defines the common parameters for the time plugin
 type Time struct {
-	// parses/formats value according to this type, default is *string
-	// +kubebuilder:validation:Enum:=float;unixtime;*string;mixed
+	// parses/formats value according to this type, default is string
+	// +kubebuilder:validation:Enum:=float;unixtime;string;mixed
 	TimeType *string `json:"timeType,omitempty"`
-	// Process value according to the specified format. This is available only when time_type is *string
+	// Process value according to the specified format. This is available only when time_type is string
```
Comment (benjaminhuo) [inline apis/fluentd/v1alpha1/plugins/common/common_types.go:26]: should change like this?
```suggestion
	// Process value according to the specified format. This is available only when time_format is string
```

**Comment by wenchajun** [comment]
https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L355
It also seems to need revision here.

**Comment by benjaminhuo** [comment]
> https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L355 It also seems to need revision here.

Yes, the CI error also needs to be fixed @Jakob3xD 
https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L556
image

**Comment by Jakob3xD** [comment]
> Yes, the CI error also needs to be fixed @Jakob3xD

Should be fixed now. I updated the key and removed the new line at the end of file which was added by my vim.

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@Jakob3xD Thank you

**Pull request merged**